webr-vars.mk

```
@@ -9,11 +9,42 @@ R_VERSION = $(shell cat $(WEBR_ROOT)/R/R-VERSION)
 R_SOURCE = $(WEBR_ROOT)/R/build/R-$(R_VERSION)
 R_HOME = $(WEBR_ROOT)/wasm/R-$(R_VERSION)/lib/R
 
-WEBR_INCLUDES = -I$(R_SOURCE)/build/include -I$(R_SOURCE)/src/include
-WEBR_LDFLAGS = -L$(WEBR_ROOT)/wasm/lib -L$(WEBR_ROOT)/wasm/R-$(R_VERSION)/lib/R/lib
-
-EM_LIBS = -s USE_LIBPNG=1
-EM_CXX_FIXES = -DRCPP_DEMANGLER_ENABLED=0 -D__STRICT_ANSI__
+WASM = $(WEBR_ROOT)/wasm
+
+EM_LIBS := -s USE_LIBPNG=1
+
+WASM_OPT := $(WASM_OPT)
+WASM_OPT += -Oz
+
+WASM_COMMON_FLAGS := $(WASM_OPT)
+WASM_COMMON_FLAGS += -fPIC
+WASM_COMMON_FLAGS += -fwasm-exceptions
+WASM_COMMON_FLAGS += -s SUPPORT_LONGJMP=wasm
+
+WASM_CFLAGS := $(WASM_CFLAGS)
+WASM_CFLAGS += $(WASM_COMMON_FLAGS)
+WASM_CFLAGS += -std=gnu11
+
+WASM_CXXFLAGS := $(WASM_CXXFLAGS)
+WASM_CXXFLAGS += $(WASM_COMMON_FLAGS)
+WASM_CXXFLAGS += -DRCPP_DEMANGLER_ENABLED=0
+WASM_CXXFLAGS += -D__STRICT_ANSI__
+
+WASM_CPPFLAGS := $(WASM_CPPFLAGS)
+WASM_CPPFLAGS += -I$(WASM)/include
+WASM_CPPFLAGS += -I$(R_SOURCE)/build/include
+WASM_CPPFLAGS += -I$(R_SOURCE)/src/include
+WASM_CPPFLAGS += $(EM_LIBS)
+
+WASM_LDFLAGS := $(WASM_LDFLAGS)
+WASM_LDFLAGS += -s SIDE_MODULE=1
+WASM_LDFLAGS += -s WASM_BIGINT
+WASM_LDFLAGS += -s ASSERTIONS=1
+WASM_LDFLAGS += -L$(WASM)/lib
+WASM_LDFLAGS += -L$(WASM)/R-$(R_VERSION)/lib/R/lib
+WASM_LDFLAGS += $(EM_LIBS)
+WASM_LDFLAGS += -fwasm-exceptions
+WASM_LDFLAGS += -s SUPPORT_LONGJMP=wasm
```

```suggestion
WASM_LDFLAGS += -s SUPPORT_LONGJMP=wasm
WASM_LDFLAGS += $(WASM_OPT)
```

Emscripten additionally takes optimisation flags at link time.